src/hotspot/share/cds/cdsConfig.cpp

```diff
@@ -643,8 +643,15 @@ bool CDSConfig::check_vm_args_consistency(bool patch_mod_javabase, bool mode_fla
   }
 
   if (is_dumping_static_archive()) {
-    if (is_dumping_preimage_static_archive() || is_dumping_final_static_archive()) {
-      // Don't tweak execution mode
+    if (is_dumping_preimage_static_archive()) {
+      // Don't tweak execution mode during AOT training run
+    } else if (is_dumping_final_static_archive()) {
+      if (Arguments::mode() == Arguments::_comp) {
+        // -Xcomp triggers blocking compilation for all called methods,
+        // but during AOT assembly phase we should AOT compile in parallel
+        // (and not blocking) only methods collected during training run.
```

Comment:
```suggestion
        // Improve testing time with AOT and -Xcomp.
        // AOT assembly phase compiles methods collected during training run
        // with blocking compilation requests. There is no reason to compile the rest
        // of the core/infrastructure methods with aggressive/blocking compilation.
```

Reply:
Thank you @shipilev for looking on this.

> with blocking compilation requests.

AOT compilation requests are not blocking. We submit bunch of requests for the same compilation level and only then wait when  queues are empty.

With -Xcomp we wait each compilation request. 

So I want to use "with non-blocking compilation requests. ". There rest of suggested comment is fine.

Reply:
Ah yes, right, d'uh. I wrote some of that code! Let me rewrite the comment...